From 9d93ca19f4ad0ca327964b6234316547d774b17f Mon Sep 17 00:00:00 2001
From: Agriya Khetarpal <74401230+agriyakhetarpal@users.noreply.github.com>
Date: Sat, 17 Aug 2024 01:12:28 +0530
Subject: [PATCH 16/19] Make `sf_error_state_lib` a static library

wasm.ld does not support linkage with shared libraries. This patch
changes `sf_error_state_lib` to a static one.

---
 scipy/special/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scipy/special/meson.build b/scipy/special/meson.build
index 82b813ea85..24bee0a21c 100644
--- a/scipy/special/meson.build
+++ b/scipy/special/meson.build
@@ -33,7 +33,7 @@ else
   scipy_import_dll_args = []
 endif
 
-sf_error_state_lib = shared_library('sf_error_state',
+sf_error_state_lib = static_library('sf_error_state',
   ['sf_error_state.c'],
   include_directories: ['../_lib', '../_build_utils/src'],
   c_args: scipy_export_dll_args,
-- 
2.39.3 (Apple Git-146)

